packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -38,30 +40,103 @@ class _TextSelectionToolbar extends StatelessWidget {
   final VoidCallback handlePaste;
   final VoidCallback handleSelectAll;
 
+  @override
+  __TextSelectionToolbarState createState() => __TextSelectionToolbarState();
+}
+
+class __TextSelectionToolbarState extends State<_TextSelectionToolbar> {
+  // effective when items > 3
+  bool flipToMenu = false;
+
   @override
   Widget build(BuildContext context) {
     final List<Widget> items = <Widget>[];
-    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-
-    if (handleCut != null)
-      items.add(FlatButton(child: Text(localizations.cutButtonLabel), onPressed: handleCut));
-    if (handleCopy != null)
-      items.add(FlatButton(child: Text(localizations.copyButtonLabel), onPressed: handleCopy));
-    if (handlePaste != null)
-      items.add(FlatButton(child: Text(localizations.pasteButtonLabel), onPressed: handlePaste,));
-    if (handleSelectAll != null)
-      items.add(FlatButton(child: Text(localizations.selectAllButtonLabel), onPressed: handleSelectAll));
+    final MaterialLocalizations localizations =
+        MaterialLocalizations.of(context);
+
+    if (widget.handleCut != null)
+      items.add(FlatButton(
+          child: Text(localizations.cutButtonLabel),
+          onPressed: widget.handleCut));
+    if (widget.handleCopy != null)
+      items.add(FlatButton(
+          child: Text(localizations.copyButtonLabel),
+          onPressed: widget.handleCopy));
+    if (widget.handlePaste != null)
+      items.add(FlatButton(
+        child: Text(localizations.pasteButtonLabel),
+        onPressed: widget.handlePaste,
+      ));
+    if (widget.handleSelectAll != null)
+      items.add(FlatButton(
+          child: Text(localizations.selectAllButtonLabel),
+          onPressed: widget.handleSelectAll));
 
     // If there is no option available, build an empty widget.
     if (items.isEmpty) {
       return Container(width: 0.0, height: 0.0);
     }
 
+    if (items.length <= 3) {
```

Comment:
The ideal solution is to adjust the number of items visible such that the maximum number fit on the screen without overflowing.  Hardcoding it to 3 is a breaking change because the many tests that depend on 4 buttons being on the screen will break.  It also may still overflow in languages with very long strings in the first 3 buttons.